src/entity/object/FallingBlock.php

```diff
@@ -153,4 +160,19 @@ protected function syncNetworkData(EntityMetadataCollection $properties) : void{
 	public function getOffsetPosition(Vector3 $vector3) : Vector3{
 		return $vector3->add(0, 0.49, 0); //TODO: check if height affects this
 	}
+
+	protected function onHitGround() : ?float{
+		if($this->getBlock()->getId() === VanillaBlocks::ANVIL()->getId()) {
```

Comment:
I still think this should be delegated to the block, similarly to how `tickFalling()` is handled.